daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DFDLXFunctions.scala

```diff
@@ -185,7 +186,7 @@ case class DFDLXTrace(recipe: CompiledDPath, msg: String)
       }
       case other: DINode => other.namedQName.toString
     }
-    System.err.println("trace " + msg + ":" + nodeString)
+    Logger.log.trace(s"dfdlx:trace ${msg} : ${nodeString}")
```

Comment:
Even though the function is called `dfdlx:trace` I'm not sure we want to actually log the message at a the `trace()` log level. The trace log level is incredibly verbose, and should really only ever be used when a developer is debugging Daffodil.

Unlike the trace log level, the `dfdlx:trace` function is used when someone wants to debug a schema, which rarely requires a deap level of daffodil debug level logging. I would argue we want to log at either the error, warning, or info log level. Those are normal log levels that won't overwhelm the user with log messages. Though, I'm not sure which is more appropriate. Info feels more correct to me, but it means the user would need to bump the verbosity to get any output since we default to the warning log level, so that might be intuitive. This would also simply the test a bit since you would really need the complex log4j2 configuration.

Maybe another option would be to use an SDW instead of the logging infrastructure. This always shows up regardless of the log level, and it can be easily suppressed. Though, there might be issue with backtracking--I'm not sure if backtracking also removes SDWs, which we definitely don't want

Reply:
I was under the impression that users would want to capture logged information by itself using the log4j configuration, but if that is not the case the level should be raised. Would it be too cost intensive to have an SDW and a log4j info?

Reply:
the level should be raised regardless and I agree info would be better